### DISCUSSION TITLE
Profile Freedom Internet

### DISCUSSION THREAD

**Discussion by Rinus-86** [fabianishere/udm-iptv#316] [Q&A]
Hello, 

I'm think of switching from Delta Fiber to Freedom Internet in the Netherlands. I was wondering if someone allready got this working with Freedom and is willing to share there profile or is there someone  who can help me creating the right profile?

**Comment by jaccobezemer** [comment]
Same here. I switched to Freedom (Canal+) and wonder what profile to use.
When I use the custom profile what IPTV_WAN_RANGES should be used.

**Comment by GuusIJ** [comment]
Today at my second evening on freedom as internet provider I managed to get iptv up and running. I followed the instructions and the shell script from fabianishere and selected the xs4all provider settings. Et voila Amino box is up and running.

In my configuration fiber comes directly over sfp to sfp ethernet which I configured with a third party gateway vlan trunk (Includes vlan 100, 6 and 4). the sfp ethernet is directly connected with wan1 port.
Internal iptv is on seperate vlan and my wan port is at vlan 6

**Reply by Rinus-86** [reply]
Hi @GuusIJ so if I you understand you correct. You just followed the instructions on the Code tab and used the xs4all profile to get the Amino working?

**Reply by jaccobezemer** [reply]
So eventually, Freedom changed my TV profile to OTT. Then the TV box works instant without VLANs, Proxies, etc. 
The only thing that doesn't work are the radio channels and teletext. Which is no problem for me.
B.t.w. I'm mostly using the Canal+ App on my Philips Android TV witch works exactly as the Amino Box.

**Reply by Rinus-86** [reply]
Hello @GuusIJ I trying it my self now but I can get it working. Is it possible that you share your profile?

**Reply by Rinus-86** [reply]
Hi I allready got is working. The xs4all profile was indeed the one I also used foor Freedom Internet.  I just forgot to disable de native IPTV option on the WAN settings in the Unifi UI.

**Reply by GuusIJ** [reply]
> Hi I allready got is working. The xs4all profile was indeed the one I also used for Freedom Internet. I just forgot to disable de native IPTV option on the WAN settings in the Unifi UI.

So I also got the amino working with xs4all profile. Live tv works, films etc works, program guide also (so the vlan 6 data streams), only the function "start from beginning" is not working. Somehow there is no stream to a proper server. I don't now how to extract the profile otherwise I would love to share.

In my udm pro I can't find the "native IPTV option"  in the settings. Perhaps you have a hint Rinus? Found it under WAN1 perhaps I need a reboot

Also I notice with some updates but not all I have to run the script again because the parameters sometimes get lost. This is the smallest problem.

**Reply by Rinus-86** [reply]
> > Hi I allready got is working. The xs4all profile was indeed the one I also used for Freedom Internet. I just forgot to disable de native IPTV option on the WAN settings in the Unifi UI.
> 
> So I also got the amino working with xs4all profile. Live tv works, films etc works, program guide also (so the vlan 6 data streams), only the function "start from beginning" is not working. Somehow there is no stream to a proper server. I don't now how to extract the profile otherwise I would love to share.
> 
> In my udm pro I can't find the "native IPTV option" in the settings. Perhaps you have a hint Rinus? Found it under WAN1 perhaps I need a reboot
> 
> Also I notice with some updates but not all I have to run the script again because the parameters sometimes get lost. This is the smallest problem.

Yes this is the same for me "start from beginning" is also not working here. Mayby if I have time I will use wireshark to check the network traffic to try and get it working. For now if I want to use it is use the app. 

The only thing I have got is that sometimes the picture freezez but I think this is due to my switch where the amino is connected to because on the other Amino i haven't had the issue which is behind an other switch.  

The Native IPTV option is under your WAN settings. It is called IPTV Streaming. 

![image](https://github.com/user-attachments/assets/b57b7c11-a33b-42ba-a676-6e96bc1df760)

By the way the profile is also shown when u use the command "udm-iptv -diag" the log shown als shows the used profile.

**Reply by GuusIJ** [reply]
In my situation I had the IPTV streaming active on wan 1 and it was functioning as described. Switched this native option off. As result of this the amino freezed a few very short times and after a while this freezing stopped. Today all works fine.

By the way I have my fiber directly on an sfp module. No modem in between so I thought that this could influence " start from beginning"  but I guess this is of no influence.

**Comment by Rinus-86** [comment]
The xs4all profile works with Freedom Internet.

**Reply by GuusIJ** [reply]
except for "start from beginning" for live tv.

I opend dhcp option 121 as instructed by freedom internet but nu succes. The start from beginning however works fine in canal+ and filmbox channels.

Configuration so far in the udm:

WAN Interface: eth8
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: br44
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
49: iptv@eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 100.68.177.224/23 scope global iptv
       valid_lft forever preferred_lft forever
100.64.0.0/10 via 100.68.176.1 metric 249 
100.68.176.0/23 proto kernel scope link src 100.68.177.224 
=== Service Logs ===
Aug 02 21:47:10 YzUDM udm-iptvd[2402]:         group time 238
Aug 02 21:47:10 YzUDM udm-iptvd[2402]:         version: 2
Aug 02 21:47:10 YzUDM udm-iptvd[2402]: interface br44:192.168.44.1
Aug 02 21:47:10 YzUDM udm-iptvd[2402]: group 224.0.251.124 exclude mode
Aug 02 21:47:10 YzUDM udm-iptvd[2402]:         group time 260
Aug 02 21:47:10 YzUDM udm-iptvd[2402]:         version: 2
Aug 02 21:47:10 YzUDM udm-iptvd[2402]: group 224.0.251.126 exclude mode
Aug 02 21:47:10 YzUDM udm-iptvd[2402]:         group time 1
Aug 02 21:47:10 YzUDM udm-iptvd[2402]:         version: 2
Aug 02 21:47:10 YzUDM udm-iptvd[2402]: group 225.0.71.1 exclude mode

**Reply by GuusIJ** [reply]
Well in the end it turned out that you don't need dhcp option 121 at all. 
Despite that all the routes are correct on the unifi pro as confirmed by freedom still no start from beginning.
So I decided to let freedom change the channel listing to OTT and everything works fine. Consequence is that there is no teletext nor radio available.

Freedom guesses that the RSTP function within the UDM isn't suitable for iptv

**Reply by Rinus-86** [reply]
Hi @GuusIJ  Does Start from beginning work in OTT? And you still use the settop box?  
Do I understand it correct that de only thing missing is teletext and Radio.

**Reply by jaccobezemer** [reply]
Hi @Rinus-86,

Yes, you still use the settop box. Start from beginning works and the only thing that is missing is teletext and radio. 
As you can read here: https://github.com/fabianishere/udm-iptv/discussions/316#discussioncomment-9605435

**Reply by Rinus-86** [reply]
@jaccobezemer  maybe a stupid question but if freedom
Changes my tv to OTT do you still need the udm-iptv service?

**Reply by jaccobezemer** [reply]
No @Rinus-86 you don't need that service or any other special configurations like vlan's and multicast.
